source/clear-linux/guides/maintenance/kernel-development.rst

```diff
@@ -0,0 +1,335 @@
+===================================================
+
+If the kernel modification you need is already open source and likely to be 
+useful to others, consider submitting a request to add or enable in the
+|CL| kernel.
+
+If your change request is accepted, you do not need to maintain your own modified kernel.
+
+Make enhancement requests to the |CL| distribution `on GitHub`_ .
+
+
+Customize the Linux kernel source
+=================================
+
+In some cases, it may be necessary modify the kernel to suit your specific 
+needs or test new kernel code as a developer.
+
+You can build and install a custom kernel, however you must:
+
+* Disable secure boot
+* Maintain any updates to the kernel going forward
+
+To customize the kernel you will need a |CL| development evnironment, 
+make changes to the kernel, build the kernel, and install it.
+
+
+Install the |CL| development tooling framework
+----------------------------------------------
+
+.. include:: autospec.rst
+   :start-after: install-tooling-after-header:
+   :end-before: install-tooling-end:
+
+
+
+Clone the Linux kernel package 
+------------------------------
+clone the existing kernel package repository from |CL| as a starting point.
+
+#. Clone the Linux kernel package from |CL|.
+
+   .. code-block:: bash
+
+      cd ~/clearlinux
+      make clone_linux
+
+#. Navigate into the cloned package directory.
+
+   .. code-block:: bash
+
+      cd ~/clearlinux/packages/linux
+
+.. note::
+    The "linux" package is the kernel that comes with |CL| in the kernel-native bundle.
+    You can alternatively use a different kernel variant as the base for modification. 
+    For a list of kernel package names which you can clone instead, see the `clearlinux-pkgs GitHub`_.
+
+
+Change the kernel version 
+-------------------------
+
+|CL| tends to use the latest kernel available from kernel.org, the Linux 
+upstream. The kernel version that will be built can be changed in the 
+RPM SPEC file.
+
+.. note:: 
+   While most packages in Clear Linux are typically packaged using 
+   :ref:`autospec`, the kernel is not. This means control files provided 
+   by autospec are not available  and changes must be made directly.
+
+#. Open the Linux kernel package RPM SPEC file in an editor.
+
+   .. code-block:: bash
+
+      $EDITOR linux.spec
+
+#. Modify the Version, Release, and Source0 URL entries at the top of the file.
+
+      .. code-block:: bash
+      
+       Name:           linux
+       Version:        4.20.5
+       Release:        690
+       License:        GPL-2.0
+       Summary:        The Linux kernel
+       Url:            http://www.kernel.org/
+       Group:          kernel
+       Source0:        https://cdn.kernel.org/pub/linux/kernel/v4.x/linux-4.20.5.tar.xz
+       Source1:        config
+       Source2:        cmdline
+
+   .. note::
+      TIP: Consider changing the Name in the RPM spec file to easily identify a modified kernel.
```

Comment:
I'm assuming you are referring the Name: field in the local copy of linux.spec shown above.  Any hard and fast rules to follow on the name?

Reply:
Right. Can you think of a way to make this more...clear?

No hard and fast rules that I am aware of. I imagine the limiting factor would be the filesystem max length filename.